### PR TITLE
gitify: init at 5.16.1

### LINKED ISSUE #250470: Package request: gitify
**Project description**

**Metadata**

* homepage URL: [www.gitify.io](https://www.gitify.io/) 
* source URL: https://github.com/gitify-app/gitify
* license: mit
* platforms: unix, linux, darwin, windows

### PR DESCRIPTION
**Pull request by pineapplehunter**
## Description of changes
Adds gitify package

closes #250470

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/gi/gitify/package.nix [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+    owner = "gitify-app";
+    repo = "gitify";
+    rev = "v${version}";
+    hash = "sha256-j67+T+2hHtzXAcbOnh3eihOYq0Fp6XxBV2Dxw/hsncg=";
+  };
+
+  nativeBuildInputs = [
+    nodejs
+    pnpm.configHook
+    copyDesktopItems
```
[the remark this answers is not in the document]
Comment (pineapplehunter) [reply pkgs/by-name/gi/gitify/package.nix:29]: It is used to copy the `desktopItems` into the output `share/application`.

File: pkgs/by-name/gi/gitify/package.nix [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+    imagemagick
+    makeWrapper
+  ];
+
+  pnpmDeps = pnpm.fetchDeps {
+    inherit pname version src;
+    hash = "sha256-oxx/z3dSnOwEhINU9nRIlw1kv6XvJnSFGKykW0ZxzwI=";
+  };
+
+  ELECTRON_SKIP_BINARY_DOWNLOAD = 1;
```
[the remark this answers is not in the document]
Comment (pineapplehunter) [reply pkgs/by-name/gi/gitify/package.nix:39]: I will fix that.
Out of curiosity, is there a difference between writing it directly in the derivation and putting it in env?

File: pkgs/by-name/gi/gitify/package.nix [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+      categories = [ "Development" ];
+      startupWMClass = "Gitify";
+    })
+  ];
+
+  passthru.updateScript = nix-update-script { };
+
+  meta = {
+    homepage = "https://www.gitify.io/";
+    description = "GitHub Notifications on your menu bar";
```
[the remark this answers is not in the document]
Comment (pineapplehunter) [reply pkgs/by-name/gi/gitify/package.nix:104]: Fixed

File: pkgs/by-name/gi/gitify/package.nix (hunk 1 above)
[the remark this answers is not in the document]
Comment (pineapplehunter) [reply pkgs/by-name/gi/gitify/package.nix:29]: I haven't checked thoroughly, but I think this is a hook that is called on postInstall.
If I delete this line, I won't get the desktop file in the output.

File: pkgs/by-name/gi/gitify/package.nix [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+}:
+
+stdenv.mkDerivation rec {
+  pname = "gitify";
+  version = "5.13.1";
+
+  src = fetchFromGitHub {
+    owner = "gitify-app";
+    repo = "gitify";
+    rev = "v${version}";
```
[the remark this answers is not in the document]
Comment (pineapplehunter) [reply pkgs/by-name/gi/gitify/package.nix]: Fixed

File: pkgs/by-name/gi/gitify/package.nix [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+  nodejs,
+  electron,
+  makeDesktopItem,
+  copyDesktopItems,
+  imagemagick,
+  makeWrapper,
+  nix-update-script,
+}:
+
+stdenv.mkDerivation rec {
```
Comment (Luflosi) [inline pkgs/by-name/gi/gitify/package.nix:15]:
```suggestion
stdenv.mkDerivation (finalAttrs: {
```

File: pkgs/by-name/gi/gitify/package.nix [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+  pname = "gitify";
+  version = "5.13.1";
+
+  src = fetchFromGitHub {
+    owner = "gitify-app";
+    repo = "gitify";
+    rev = "refs/tags/v${version}";
```
Comment (Luflosi) [inline pkgs/by-name/gi/gitify/package.nix:22]:
```suggestion
    rev = "refs/tags/v${finalAttrs.version}";
```

File: pkgs/by-name/gi/gitify/package.nix [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+  nativeBuildInputs = [
+    nodejs
+    pnpm.configHook
+    copyDesktopItems
+    imagemagick
+    makeWrapper
+  ];
+
+  pnpmDeps = pnpm.fetchDeps {
+    inherit pname version src;
```
Comment (Luflosi) [inline pkgs/by-name/gi/gitify/package.nix:35]:
```suggestion
    inherit (finalAttrs) pname version src;
```

File: pkgs/by-name/gi/gitify/package.nix [hunk 8]
```diff
@@ -0,0 +1,109 @@
...
+
+  meta = {
+    homepage = "https://www.gitify.io/";
+    changelog = "https://github.com/gitify-app/gitify/releases/tag/v${version}";
+    description = "GitHub Notifications on your menu bar";
+    license = lib.licenses.mit;
+    maintainers = with lib.maintainers; [ pineapplehunter ];
+    platforms = lib.platforms.all;
+  };
+}
```
Comment (Luflosi) [inline pkgs/by-name/gi/gitify/package.nix:109]:
```suggestion
})
```

**Review by Luflosi — CHANGES_REQUESTED** [review]
How about using the `finalAttrs` pattern?

**Comment by pineapplehunter** [comment]
@Luflosi Thanks for the suggestion!
I have updated the commit.

**Review by Luflosi — APPROVED** [review]
The code looks good.

**Comment by Luflosi** [comment]
I used to use this app on macOS. How can I run it on Linux? I tried simply executing `bin/gitify` but nothing seems to happen. Maybe it doesn't work with GNOME? Or do I need to actually install it?

**Comment by pineapplehunter** [comment]
I'm using it on gnome now, so I'm not sure what the issue is..
When I first ran it, it only showed a status icon (you may need a gnome extension to show the status icons). Double-clicking the icon showed the application window.

**Comment by Luflosi** [comment]
Thanks for the hint. I got it working by installing `gnomeExtensions.appindicator`.

**Comment by pineapplehunter** [comment]
Updated the package since
- there was a new version available
- the update to pnpm was causing the pnpmDeps hash to change

**Review by Luflosi — APPROVED** [review]

File: pkgs/by-name/gi/gitify/package.nix [hunk 9]
```diff
@@ -0,0 +1,109 @@
...
+        -c.electronVersion="${electron.version}" \
+
+    runHook postBuild
+  '';
+
+  installPhase = ''
+    runHook preInstall
+
+    ${
+      if stdenv.isDarwin then
```
Comment (FliegendeWurst) [inline pkgs/by-name/gi/gitify/package.nix:61]:
```suggestion
      if stdenv.hostPlatform.isDarwin then
```

File: pkgs/by-name/gi/gitify/package.nix [hunk 10]
```diff
@@ -0,0 +1,109 @@
...
+        ''
+      else
+        ''
+          mkdir -p $out/share/gitify
+          cp -r dist/*-unpacked/{locales,resources{,.pak}} $out/share/gitify
+
+          for size in 16 32 48 64 128 256 512 1024; do
+            mkdir -p $out/share/icons/hicolor/"$size"x"$size"/apps
+            magick assets/images/app-icon.ico -resize "$size"x"$size" $out/share/icons/hicolor/"$size"x"$size"/apps/gitify.png
+          done
```
Comment (FliegendeWurst) [inline pkgs/by-name/gi/gitify/package.nix:75]: I think it is enough to install one size

Comment (pineapplehunter) [reply pkgs/by-name/gi/gitify/package.nix]: Thanks for the review!
I will leave only the size of the original icon, which is 256x256.

**Comment by pineapplehunter** [comment]
The build is failing on x86_64-darwin.
I will update the commit when I fix the issue.

**Review by FliegendeWurst — APPROVED** [review]

**Review by Aleksanaa — APPROVED** [review]

**Pull request merged**